src/sagas/RootSaga.test.js

```diff
@@ -0,0 +1,121 @@
+      // ... and done.
+      expect(gen.next().value).toBeUndefined();
+    });
+  });
+
+  describe('saveUserProfile', () => {
+    it('should successfully process a saveUserProfile request if there are no exceptions', () => {
+      const service = new MockUserAccountApiService();
+      const rootGen = rootSaga(service);
+      rootGen.next(); // Causes the service to be set.
+
+      const action = profileActions.saveUserProfile(
+        'my username',
+        {
+          fullName: 'Full Name',
+          education: 'b',
+        },
+        'ze field',
+      );
+      const gen = saveUserProfile(action);
+      const userAccount = {
+        name: 'Full Name',
+        levelOfEducation: 'b',
+      };
+      expect(gen.next().value).toEqual(put(profileActions.saveUserProfileBegin()));
+      expect(gen.next().value).toEqual(call(service.saveUserAccount, 'my username', userAccount));
+      // The library would supply the result of the above call
+      // as the parameter to the NEXT yield.  Here:
+      expect(gen.next(userAccount).value).toEqual(put(profileActions.saveUserProfileSuccess()));
+      expect(gen.next().value).toEqual(put({
+        type: 'FETCH_USER_ACCOUNT_SUCCESS',
+        payload: { userAccount },
+      }));
+      expect(gen.next().value).toEqual(delay(300));
+      expect(gen.next().value).toEqual(put(profileActions.closeEditableField('ze field')));
+      expect(gen.next().value).toEqual(delay(300));
+      expect(gen.next().value).toEqual(put(profileActions.saveUserProfileReset()));
+      expect(gen.next().value).toBeUndefined();
+    });
+
+    it('should successfully publish a failure action on exception', () => {
+      const service = new MockUserAccountApiService();
+      const error = new Error('uhoh');
+      const rootGen = rootSaga(service);
+      rootGen.next(); // Causes the service to be set.
+
+      const action = profileActions.saveUserProfile(
+        'my username',
+        {
+          fullName: 'Full Name',
+          education: 'b',
+        },
+        'ze field',
+      );
+      const gen = saveUserProfile(action);
+
+      expect(gen.next().value).toEqual(put(profileActions.saveUserProfileBegin()));
+      const result = gen.throw(error);
+      expect(result.value).toEqual(put(profileActions.saveUserProfileFailure(error)));
+      expect(gen.next().value).toBeUndefined();
+    });
+  });
+
+  describe('mapDataForRequest', () => {
+    it('should modify props according to prop modifier strings and functions', () => {
+      const props = {
+        favoriteColor: 'red',
+        age: 30,
+        petName: 'Donkey',
+        fullName: 'Donkey McWafflebatter',
+        userLocation: 'US',
+        education: 'BS',
+        socialLinks: {
+          twitter: {
+            url: 'https://www.twitter.com',
+          },
+          facebook: {
+            url: 'https://www.facebook.com',
+          },
+        },
+      };
```

Comment:
Try rebasing from master now that social links saving is fixed. The PROP_TO_STATE_MAP in RootSaga.js is no long expecting an object of socialLinks, but an array – mirroring the API.

Reply:
👍 Just figured this out. :) Thanks!